pymodbus/server/simulator/http_server.py

```diff
@@ -154,7 +154,15 @@ def __init__(
         self.datastore_context = ModbusSimulatorContext(
             device, custom_actions_dict or {}
         )
-        datastore = ModbusServerContext(slaves=self.datastore_context, single=True)
+        datastore = None
+        if "unit_address" in server:
+            # Designated ModBus unit address. Will only serve data if the address matches
+            address = int(server["unit_address"])
```

Comment:
No need to make a local variable, with a confusing name, combine the lines.

Reply:
Fixed